src/components/svelte-web.ts

```diff
@@ -134,8 +134,11 @@ export default function registerWebComponent(
       const slotsNames = Array.from(this.children).map((c) =>
         c.getAttribute('slot')
       )
-      // Add default slot if there are nodes without a slot name.
-      if (this.childNodes.length > slotsNames.length) slotsNames.push(null)
+      // Add default slot if there are non-empty nodes without a slot name.
+      const nonEmptyNodes = Array.from(this.childNodes).filter(
```

Comment:
`this.childNodes` will always be nodes without a slot name?

Reply:
Nah, `childNodes` includes all childNodes of the element (including `text` nodes, which can't have a slot attribute).

This is basically to handle the case where the slotted content is `#text` (which won't show up in the `children` array) but which indicates we have a default slot.

This was working well before, but Lit is slotting whitespace into the element which means we're adding a default slot to our component (which in this case, swaps the icon to child element mode, and doesn't render the named icon).